internal/policygen/iam.go

```diff
@@ -15,62 +15,103 @@
 package policygen
 
 import (
+	"encoding/json"
 	"fmt"
-	"log"
+	"os/exec"
+	"path/filepath"
+	"strings"
 
+	"github.com/GoogleCloudPlatform/healthcare-data-protection-suite/internal/runner"
+	"github.com/GoogleCloudPlatform/healthcare-data-protection-suite/internal/template"
 	"github.com/GoogleCloudPlatform/healthcare-data-protection-suite/internal/terraform"
 	"github.com/hashicorp/terraform/states"
 )
 
-type binding struct {
-	Type   string
-	ID     string
-	Role   string
-	Member string
+type root struct {
+	Type string
+	ID   string
 }
 
-func generateIAMBindingsPolicies(resources []*states.Resource, outputPath string) error {
+// All IAM members associated with a single role.
+type roleBindings map[string][]string
 
-	bindings, err := bindings(resources)
-	if err != nil {
+// All roles associated with a root resource (organization, folder or project)
+var bindings = make(map[root]roleBindings)
```

Comment:
I find it smelly that this is a global variable (sort of) and is being secretly modified by `allBindings`. Why not have `allBindings` generate and return it?

Reply:
Done